packages/styles/src/core/variables/_variables.scss

```diff
@@ -1,3 +1,9 @@
 @import '_colors';
 @import '_breakpoints';
 @import '_sizes';
+
+$u1: 4px;
+$u2: 2 * $u1; // 8
+$u3: 2 * $u2; // 16
+$u4: 2 * $u3; // 32
+$u5: 2 * $u4; // 64
```

Comment:
I don't get making the `$u` exponential.

The grid is 4px based. Where are 12px, 20px, 40px... ?

Reply:
otherwise we would need 64 variables...

Reply:
I'm pretty sure that the low ones (4, 8, 12, 16, 20, 24) will be way more frequent that 32 or 64. This exponential growth doesn't reflect the grid steps.

And when doing tiny adjustments you'll go from one value to the nearest one.

Go for
4 = u1 - 1 space
8 = u2 - 2 spaces
12 = u3 - 3 spaces
16 = u4 - 4 spaces

instead of
4 = u1 - 1 space
8 = u2 - 2 spaces
16 = u3 - 4 spaces
32 = u4 - 8 spaces
64 = u5 - 16 spaces

There's no visual relation between the u suffix and the real grid steps.

Reply:
I'd go for one base unit variable, such as `$su` (I talked with iago about this), and then multiply wherever we need. 

So that `$toolbar-size: 52px;` would be `$toolbar-size: 13 * $su;`